### ISSUE TITLE
Retry on Failure

### ISSUE THREAD

**Issue by InteXX** [ohadschn/letsencrypt-webapp-renewer#114]
Is there a way to retry if the process fails for any reason?

I encountered this error:

> [09/01/2020 00:01:37 > dd0b6d: INFO] Error during Request or install certificate System.Exception: Unable to complete challenge with Lets Encrypt servers error was: {"type":"http-01","url":"https://acme-v02.api.letsencrypt.org/acme/chall-v3/6903145984/zVwXJQ","status":"Invalid","validated":null,"error":{"Type":"urn:ietf:params:acme:error:dns","Detail":"During secondary validation: No valid IP addresses found for [Redacted]","Identifier":null,"Subproblems":null,"Status":400},"errors":null,"token":"[Redacted]","keyAuthorization":null}

This appears to have been a transient network problem that likely would not have repeated on a second run. Indeed, I ran the job manually just now and it completed easily and successfully.

Would you consider integrating Polly for implementing a retry policy?

**Comment by ohadschn** [comment]
Good idea, I've been running into transient failures as well, will do when I get the time.

**Comment by InteXX** [comment]
If your time is anything like mine, it's going to be hard to find ;-)
